core/src/com/unciv/logic/map/TileInfo.kt

```diff
@@ -106,10 +105,10 @@ open class TileInfo {
     // and the neighbors of a tile never change, it's much more efficient to save the list once and for all!
     @Transient private var internalNeighbors : List<TileInfo>?=null
     val neighbors: List<TileInfo>
-        get(){
-            if(internalNeighbors==null)
-                internalNeighbors = getTilesAtDistance(1)
-            return internalNeighbors!!
+        get() {
+            val ret = internalNeighbors ?: getTilesAtDistance(1).toList()
```

Comment:
I think out was more readable before - do we gain something from this change? Maybe I'm missing something?

Reply:
Oh, this is one place that elvis operator is very useful.  You have to use the bang (`!!`) when you return while my code does not.  The type of ret (which I can rename for more clarity if you want) is `List<TileInfo>` while your type is `List<TileInfo>?` as a result, I can safely return mine while setting internalNeighbors to not be null.

However, in my latest branch, this is switched out for the kotlin lazy construct instead which makes this whole part moot. But this is a good example of useful elvis operator.